src/main/java/com/atlassian/bitbucket/jenkins/internal/trigger/BitbucketWebhookConsumer.java

```diff
@@ -249,9 +247,9 @@ public Map<SCMHead, SCMRevision> heads(SCMSource source) {
                 return emptyMap();
             }
 
-            BitbucketPullRequestRef fromRef = getPayload().getPullRequest().getFromRef();
-            return Collections.singletonMap(new GitBranchSCMHead(fromRef.getDisplayId()),
-                    new GitBranchSCMRevision(new GitBranchSCMHead(fromRef.getDisplayId()), fromRef.getLatestCommit()));
+            BitbucketPullRequestSCMHead prScmHead = new BitbucketPullRequestSCMHead(getPayload().getPullRequest());
+            return Collections.singletonMap(prScmHead,
```

Comment:
This is totally fine, but we also have `Map.of(k,v)` which may be nicer? Up to you.